src/ripe_rainbow/domain/logic/ripe_retail.py

```diff
@@ -112,6 +114,11 @@ def set_part(
             ".pickers .button-part > p:not(.no-part)",
             condition = lambda e: e.text == part.upper()
         )
+
+        # the color transition must finish before we can keep going,
+        # otherwise scrolling to the correct color won't work
+        time.sleep(1)

```

Comment:
Why is this necessary? click_when_possible should guarantee click only when the swatch is visible

Reply:
Because "when possible" = "is visible" (display != none & opacity > 0), which in this case is not enough.

Reply:
We must fix the base code then

Reply:
When I answered I thought this was about something else. The problem here is what says in the comment, without the sleep this might sometimes happen:
1. Click part
2. Retail starts scrolling the colors to show the right color for that part
3. Because Rainbow doesn't wait for step 2 to finish it tries to click a color
4. Clicking a color triggers the scrolling to that color
5. Somehow the two scrolls conflict and the prevailing one is from step 2, resulting in the click for color selection not working